Develop

gzvoznikova opened the pull request.

Commit:
sprint 5

Commit:
sprint 5

AlexKitIt commented on conftest.py:
```diff
@@ -0,0 +1,13 @@
+@pytest.fixture(scope="function")
+def driver():
+    driver = webdriver.Chrome(options=webdriver.ChromeOptions())
+    driver.get("https://stellarburgers.nomoreparties.site/")
```
Можно лучше: урлы стоит хранить отдельно, они постоянно переиспользуются, для них стоит создать модуль urls или отдельный класс в data

AlexKitIt commented on locators.py:
```diff
@@ -0,0 +1,80 @@
+    selected_button = By.XPATH, ('//div[@class = "tab_tab__1SPyG tab_tab_type_current__2BEPc pt-4 pr-10 pb-4 pl-10 noselect"]')
+
+    # Заголовок раздела "Булки" в меню конструктора
+    bread_block = By.XPATH, '/html/body/div/div/main/section[1]/div[1]/div[1]/span[text() = "Булки"]'
+
+    # Заголовок раздела "Соусы" в меню конструктора
+    sause_block = By.XPATH, '/html/body/div/div/main/section[1]/div[1]/div[2]/span[text() = "Соусы"]'
+
+    # Заголовок раздела "Начинки" в меню конструктора
+    topping_block = By.XPATH, '/html/body/div/div/main/section[1]/div[1]/div[3]/span[text() = "Начинки"]'
+
+    # Кликабельный логотип в шапке сайта
+    logo = By.XPATH, '/html/body/div/div/header/nav/div[@class="AppHeader_header__logo__2D0X2"]'
```
Нужно исправить: здесь и далее: использование порядкового номера элемента в dom является весьма нестабильным способом поиска, необходимо привязываться к атрибутам конкретного элемента
Кроме того ходить от html и использовать абсолютные пути нельзя - очень ненадежно, добавится на пути новый элемент - тест ляжет. Все подобные локаторы необходимо переписать

AlexKitIt commented on tests/test_reg.py:
```diff
@@ -0,0 +1,71 @@
+    # Регистрация аккаунта пользователя с валидными значениями
+    def test_reg_new_account_success(self, driver):
+        username = 'lina111111'
+        email = 'lina111111@ya.ru'
```
Нужно исправить: для теста регистрации email должен генерироваться каждый раз новый, в противном случае мы получим ошибку уже созданного пользователя. Необходимо написать метод, генерирующий email и положить его в helpers

Commit:
внесены правки:
1. для урла создан urls и отдельный класс DataUrls
2. переписаны локаторы  к атрибутам конкретного элемента
3. каждый раз генерируется новый email

gzvoznikova committed:
Update conftest.py
